spack/packages/dla-future/package.py

```diff
@@ -16,23 +15,41 @@ class DlaFuture(CMakePackage):
 
     version('develop', branch='master')
 
-    variant('gpu', default=False,
+    variant('cuda', default=False,
             description='Use the GPU/cuBLAS back end.')
-
-    # Until mpich is default comment this out
+    variant('test', default=False,
+            description='Run unit tests.')
+    variant('doc', default=False,
+            description='Build documentation.')
+    
     #depends_on('mpi@3:')
-    depends_on('mpich')
-    depends_on('mkl')
+    depends_on('mpi')
+    depends_on('blas')
+    depends_on('lapack')
     depends_on('blaspp')
     depends_on('lapackpp')
-    depends_on('hpx cxxstd=14 networking=none')
-    depends_on('cuda', when='gpu=True')
+    depends_on('hpx@1.4.0 cxxstd=14 networking=none')
+    depends_on('cuda', when='cuda=True')
 
     def cmake_args(self):
        spec = self.spec
-       args = ['-DDLAF_WITH_MKL=ON']
-
-       if '+gpu' in spec:
+       
+       if (spec.satisfies('^intel-mkl')):
+           args = ['-DDLAF_WITH_MKL=ON']
+       else:
+           args = ['-DDLAF_WITH_MKL=OFF']
+           args.append('-DLAPACK_TYPE=Custom')
+           args.append('-DLAPACK_LIBRARY=-L{0} -lblas -llapack'.format(spec['lapack'].prefix.lib))
+
+       if '+cuda' in spec:
            args.append('-DDLAF_WITH_CUDA=ON')
+           args.append('-DCUDA_TOOLKIT_ROOT_DIR:STRING={0}'.format(spec['cuda'].prefix))
```

Comment:
Are we sure we need this line? Recent versions of cuda's spack package set the `CUDA_HOME` variable when building, would that be sufficient?

Reply:
You're right, I've tried to substitute the -DCUDA_TOOLKIT_ROOT_DIR with -DCUDA_HOME and it does compile (it does not pass tests but maybe we can think about that later). I'm going to fix it, thanks!

Reply:
Here I was actually wondering whether it will work without either `CUDA_HOME` or `CUDA_TOOLKIT_ROOT_DIR`? I meant to say that with recent spack versions, the `cuda` package sets the `CUDA_HOME` variable at build time, if that is the case, there shouldn't be a need to set it in our `package.py`. Would you be able to confirm if this is true?

Reply:
I'm still not sure it will work on Power cluster, but as soon as we close the PR #197 I will try it.